packages/flutter/test/widgets/grid_view_test.dart

```diff
@@ -899,4 +899,62 @@ void main() {
     expect(controller.position.maxScrollExtent, 472.0);
     expect(controller.position.pixels, 472.0);
   });
+
+  testWidgets('SliverGridDelegateWithFixedCrossAxisCount mainAxisExtent corrected', (WidgetTester tester) async {
```

Comment:
Can you add to each of these tests an actual grid of children? I think in the case where childMainAxisExtent is  0 there could be a crash through a divide by zero error in SliverGridRegularTileLayout. This would only become apparent when building the grid. Right now these tests only verify the value of childMainAxisExtent. :)